app/features/recent-list/actionTypes.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * The type of (redux) action that saves the current conference details.
+ *
+ * {
+ *     type: SAVE_CURRENT_CONFERENCE,
+ *     conference: Object
+ * }
+ */
+export const SAVE_CURRENT_CONFERENCE = Symbol('SAVE_CURRENT_CONFERENCE');
+
+/**
+ * The type of (redux) action that updates the endtime of the last conference.
+ *
+ * {
+ *     type: UPDATE_END_TIME
+ * }
+ */
+export const UPDATE_END_TIME = Symbol('UPDATE_END_TIME');
```

Comment:
Because we want to store things, right? So, if you look at it, the CONFERENCE_JOINED action already encapsulates what you need to store it. In fact you pass along the same `conference` object.

It's similar for CONFERENCE_ENDED.

Reply:
> We are storing things through the recent-list reducer. Do you want me to move functionality here?

Yep

Reply:
Then we won't have a reducer there. Shouldn't this functionality be in `features/recent-list`?

Since the motive was to gain the flexibility to do more things when the conference ends (or starts) wouldn't it be better to dispatch actions of other features instead of implementing every feature's functionality here?

Reply:
> Then we won't have a reducer there. Shouldn't this functionality be in features/recent-list?

We can have multiple reducers for the same actions. The recent-list feature will have a reducer for the conference actions and store recent list related data.



> Since the motive was to gain the flexibility to do more things when the conference ends (or starts) wouldn't it be better to dispatch actions of other features instead of implementing every feature's functionality here?

I'm not sure I understand, can you give an example or rephrase?

Reply:
> We can have multiple reducers for the same actions. The recent-list feature will have a reducer for the conference actions and store recent list related data.

Ooh. So I'll remove the actions in recent-list and remove the middleware in the conference.

> I'm not sure I understand, can you give an example or rephrase?

I was thinking when CONFERENCE_JOINED is dispatched we'll dispatch all actions we need to at that time using middleware. But since we can have multiple reducers for the same actions, this is not needed anymore. :sweat_smile: